paymentapi/src/main/java/fi/hel/verkkokauppa/payment/api/PaytrailPaymentController.java

```diff
@@ -193,4 +193,56 @@ public ResponseEntity<TreeMap<String, String>> getCardFormParameters(
             throw new CommonApiException(HttpStatus.INTERNAL_SERVER_ERROR, error);
         }
     }
+
+    @GetMapping("/subscription/get/update-card-form-parameters")
+    public ResponseEntity<TreeMap<String, String>> getUpdateCardFormParameters(
+            @RequestParam(value = "merchantId") String merchantId,
+            @RequestParam(value = "namespace") String namespace,
+            @RequestParam(value = "orderId") String orderId
+    ) {
+        try {
+            String paytrailMerchantId = commonServiceConfigurationClient.getMerchantConfigurationValue(merchantId, namespace, ServiceConfigurationKeys.MERCHANT_PAYTRAIL_MERCHANT_ID);
```

Comment:
could these paytrail needed values be wrapped in some kind of  service that handles validation etc and return values.

Reply:
Example can be found here, and even a good function to be used https://github.com/City-of-Helsinki/verkkokauppa-core/blob/10f046b7cd5ef004cc1e7688ca16108e34178445/paymentapi/src/main/java/fi/hel/verkkokauppa/payment/paytrail/context/PaytrailPaymentContextBuilder.java#L33